### PR TITLE
Usage sample of BYOD with ChatCompletions

### PR DESCRIPTION
**Pull request by brandom-msft**
Resubmitting PR, now against the latest merged TSP and Java changes.

### CODE REVIEW

**Review by jpalvarezl — APPROVED** [review]
Looks good!

**Comment by jpalvarezl** [comment]
I think that if you add the comments with the links (a bit of a nit-pick from my side) and we merge, we are good to go with the deployment of `beta.4`. Wdyt @mssfang ?

File: sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/usage/ChatCompletionsWithYourData.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import com.azure.core.util.BinaryData;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Sample demonstrates the "Azure OpenAI on your data" feature.
+ */
+public class ChatCompletionsWithYourData {
```
Comment (mssfang) [inline sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/usage/ChatCompletionsWithYourData.java:26]: should we add an async sample as well?

**Review by mssfang — APPROVED** [review]

File: sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/usage/ChatCompletionsWithYourData.java (hunk 1 above)
Comment (brandom-msft) [reply sdk/openai/azure-ai-openai/src/samples/java/com/azure/ai/openai/usage/ChatCompletionsWithYourData.java:26]: Yeah, I think it is useful for us to show the different call paradigms and even usage scenarios (e.g. with search index versus without). When I was adding this one, I couldn't help but feel a growing overlap of our "usage" samples with our unit tests - I think it kind of "waters down" the utility of each, if that makes sense. I'll dedicate some time next week to sorting this out (and when I do, I'll make sure we have coverage for things like async vs sync like you mentioned)

**Comment by brandom-msft** [comment]
> I think that if you add the comments with the links (a bit of a nit-pick from my side) and we merge, we are good to go with the deployment of `beta.4`. Wdyt @mssfang ?

Not nitpicky at all, this is all new stuff for customers - it's a great suggestion.

**Pull request merged**